### ISSUE TITLE
clippy::cast_possible_truncation - false reports on statically calculated constants

### ISSUE THREAD

**Issue by mikea** [rust-lang/rust-clippy#15607]
### Summary

There is no platform in the universe for which this constitutes a loss a precision.
It is unbelievably noisy in ffi environment because of that.

### Lint Name

cast_possible_truncation

### Reproducer

I tried this code:

```rust
```
#[deny(clippy::cast_possible_truncation)]
fn main() {
    let x = std::mem::size_of::<i32>() as u32;
    println!("{x}");
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=313f3e61369b5e35e4f8f830b53ea026

### Version

```text
clippy: 0.1.89 (2025-08-04 29483883ee)
```

**Comment by KisaragiEffective** [comment]
It isn't falsy-flagged, but this special case could be improved because the return value is predictable for clippy by only looking type variable.
